File: compile-template.go
package main

import (
	"encoding/json"
	"flag"
	"fmt"
	"os"
	"regexp"
	//"strings"
)

/**
 * Script per compilare il template WordPress con i dati del cliente
 *
 * Utilizzo:
 * go run compile-template.go -template template.md -data data.json -output output.md
 */

func main() {
	// Definizione dei flag della riga di comando
	templatePath := flag.String("template", "", "Percorso del file template")
	dataPath := flag.String("data", "", "Percorso del file JSON con i dati")
	outputPath := flag.String("output", "", "Percorso del file di output")

	// Parsing degli argomenti
	flag.Parse()

	// Verifica che tutti i parametri necessari siano stati forniti
	if *templatePath == "" || *dataPath == "" || *outputPath == "" {
		fmt.Println("Utilizzo: go run compile-template.go -template template.md -data data.json -output output.md")
		flag.PrintDefaults()
		os.Exit(1)
	}

	// Lettura del file template
	templateContent, err := os.ReadFile(*templatePath)
	if err != nil {
		fmt.Printf("Errore: Impossibile leggere il file template - %v\n", err)
		os.Exit(1)
	}

	// Lettura del file JSON
	dataContent, err := os.ReadFile(*dataPath)
	if err != nil {
		fmt.Printf("Errore: Impossibile leggere il file dati - %v\n", err)
		os.Exit(1)
	}

	// Parsing del JSON
	var data map[string]interface{}
	if err := json.Unmarshal(dataContent, &data); err != nil {
		fmt.Println("Errore: Il file JSON contiene errori di sintassi")
		os.Exit(1)
	}

	// Compila il template
	compiledContent := string(templateContent)

	// Sostituisci tutte le variabili
	for key, value := range data {
		// Converti il valore in stringa
		strValue := fmt.Sprintf("%v", value)
		
		// Crea il pattern regex per trovare {{key}}
		regex := regexp.MustCompile(`{{\s*` + regexp.QuoteMeta(key) + `\s*}}`)
		
		// Sostituisci tutte le occorrenze
		compiledContent = regex.ReplaceAllString(compiledContent, strValue)
	}

	// Scrivi il file output
	err = os.WriteFile(*outputPath, []byte(compiledContent), 0644)
	if err != nil {
		fmt.Printf("Errore: Impossibile scrivere il file di output - %v\n", err)
		os.Exit(1)
	}

	fmt.Printf("Template compilato con successo: %s\n", *outputPath)
}

File: compile-template.js
#!/usr/bin/env node

/**
 * Script per compilare il template WordPress con i dati del cliente
 * 
 * Utilizzo:
 * node compile-template.js template.md data.json output.md
 */

const fs = require('fs');
const path = require('path');

// Verifica argomenti
if (process.argv.length < 5) {
    console.error('Utilizzo: node compile-template.js template.md data.json output.md');
    process.exit(1);
}

// Percorsi dei file
const templatePath = process.argv[2];
const dataPath = process.argv[3];
const outputPath = process.argv[4];

// Leggi i file
try {
    const templateContent = fs.readFileSync(templatePath, 'utf8');
    const dataContent = fs.readFileSync(dataPath, 'utf8');
    
    // Parsing JSON
    const data = JSON.parse(dataContent);
    
    // Compila il template
    let compiledContent = templateContent;
    
    // Sostituisci tutte le variabili
    for (const [key, value] of Object.entries(data)) {
        const regex = new RegExp(`{{\\s*${key}\\s*}}`, 'g');
        compiledContent = compiledContent.replace(regex, value);
    }
    
    // Scrivi il file output
    fs.writeFileSync(outputPath, compiledContent);
    
    console.log(`Template compilato con successo: ${outputPath}`);
    
} catch (error) {
    if (error.code === 'ENOENT') {
        console.error(`Errore: File non trovato - ${error.path}`);
    } else if (error instanceof SyntaxError) {
        console.error('Errore: Il file JSON contiene errori di sintassi');
    } else {
        console.error(`Errore: ${error.message}`);
    }
    process.exit(1);
}